Chessington.GameEngine/Pieces/Knight.cs

```diff
@@ -10,7 +10,143 @@ public Knight(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var moves = Enumerable.Empty<Square>();
+            var piece = board.FindPiece(this);
+
+            
```

Comment:
King and knights can at any given time move to a maximum of 8 combinations of locations. Mapping them accordingly and just checking those indexes could have reduced the logic used.

Reply:
this is true 👀

Reply:
This would help with the if branches that get quite hard and unintuitive to follow after some depth.